File: freecodecamp/bonfire/10-truncate-a-string.rb
def truncate(str, num)
  # p str.length
  if num < str.length
    num = num > 4 ? num - 3 : num
    str.slice(0, num) + '...'
  else
    str
  end
end

p truncate('A-tisket a-tasket A green and yellow basket', 11)
p truncate('Peter Piper picked a peck of pickled peppers', 14)
p truncate('A-tisket a-tasket A green and yellow basket', 'A-tisket a-tasket A green and yellow basket'.length)
p truncate('A-tisket a-tasket A green and yellow basket', 'A-tisket a-tasket A green and yellow basket'.length + 2)
p truncate('A-', 1)
p truncate("Absolutely Longer", 2)

File: freecodecamp/bonfire/10-truncate-a-string.js
function truncate(str, num) {
  if (str.length > num) {
    var newNum = num > 4 ? num - 3 : num;
    return str.slice(0, newNum) + '...';
  } else {
    return str;
  }
}

console.log(truncate('A-tisket a-tasket A green and yellow basket', 11));
console.log(truncate('Peter Piper picked a peck of pickled peppers', 14));
console.log(truncate('A-tisket a-tasket A green and yellow basket', 'A-tisket a-tasket A green and yellow basket'.length));
console.log(truncate('A-tisket a-tasket A green and yellow basket', 'A-tisket a-tasket A green and yellow basket'.length + 2));
console.log(truncate('A-', 1));
console.log(truncate("Absolutely Longer", 2));
